baseplate/ratelimit/backends/__init__.py

```diff
@@ -3,9 +3,22 @@
 from __future__ import print_function
 from __future__ import unicode_literals
 
-from .base import RateLimitBackend
+import time
 
 
-__all__ = [
-    'RateLimitBackend',
-]
+class RateLimitBackend(object):
+    """An interface for rate limit backends to implement.
+
+    :param str key: The name of the rate limit bucket to consume from.
+    :param int amount: The amount to consume from the rate limit bucket.
+    :param int allowance: The maximum allowance for the rate limit bucket.
+    :param int interval: The interval to reset the allowance.
+
+    """
+    def consume(self, key, amount, max, bucket_size):
+        raise NotImplementedError
+
+
+def _get_current_bucket(bucket_size):
```

Comment:
This should probably not have the `_` since it is meant to be imported.

Reply:
I don't think this is meant to be imported from outside of the library. It just gets imported by the two backends because it's a bit of common logic that didn't make it into the interface. I don't have a strong opinion on this though. I can drop the `_` if you want me to.

Reply:
I figured this might be used by folks who want to write their own, custom back-ends?  Maybe/maybe not?  I don't feel super strongly either way.

Reply:
eh, they can re-implement it simply for their backend if desired. i'd say keep it private for now and we can open it later if really needed.